src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,236 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../../types/Comment';
+import { addComment } from '../../api/posts';
+
+type Props = {
+  setIsErrorComments: (v: boolean) => void,
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>
+  postId: number,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  setIsErrorComments,
+  setComments,
+  postId,
+}) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const [name, setName] = useState<string>('');
+  const [isNameError, setIsNameError] = useState<boolean>(false);
+
+  const [email, setEmail] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>('Email is required');
+  const [isEmailError, setIsEmailError] = useState<boolean>(false);
+
+  const [commentText, setCommentText] = useState<string>('');
+  const [isCommentError, setIsCommentError] = useState<boolean>(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setErrorMessage('Email is required');
+    setIsLoading(true);
+
+    setIsNameError(!name);
+    setIsEmailError(!email);
+    setIsCommentError(!commentText);
+
+    if (!name || !email || !commentText) {
```

Comment:
you should trim values (I can add comment with spaces only in name and comment fields):
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/86923089/11f5c381-be02-4fc5-bda7-62f50395ffd9)